### ISSUE TITLE
Wrong expected SFT filenames when duration is not multiple of Tsft (e.g. for Tsft=1024s in many examples)

### ISSUE THREAD

**Issue by Rodrigo-Tenorio** [PyFstat/PyFstat#202]
The following happens when `Tsft` is set to `1024` instead of `1800` in the `twoF_cumulative` example.
```
(pyfstat) [rodrigo@spinning-quadrupole other_examples]$ python PyFstat_example_twoF_cumulative.py
11:05 INFO    : Creating BinaryModulatedWriter object...
11:05 INFO    : Generating SFTs with fmin=28.0, Band=4
11:05 INFO    : Checking if we can re-use injection config file...
11:05 INFO    : ...no config file PyFstat_example_data/PyFstat_example_twoF_cumulative/PyFstat_example_twoF_cumulative.cff found.
11:05 INFO    : Writing config file: PyFstat_example_data/PyFstat_example_twoF_cumulative/PyFstat_example_twoF_cumulative.cff
11:05 INFO    : Checking if we can re-use existing SFT data file(s)...
11:05 INFO    : ...no SFT file matching 'PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft' found. Will create new SFT file(s).
11:05 INFO    : Now executing: lalapps_Makefakedata_v5 --outSingleSFT=TRUE --outSFTdir="PyFstat_example_data/PyFstat_example_twoF_cumulative" --outLabel="PyFstat_example_twoF_cumulative" --IFOs="H1","L1" --sqrtSX="1e-23" --startTime=1000000000 --duration=864000 --fmin=28 --Band=4 --Tsft=1024 --injectionSources="PyFstat_example_data/PyFstat_example_twoF_cumulative/PyFstat_example_twoF_cumulative.cff" --ephemEarth="/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz" --ephemSun="/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz"
11:05 INFO    : Successfully wrote SFTs to: PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft;PyFstat_example_data/PyFstat_example_twoF_cumulative/L-843_L1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft
11:05 INFO    : Now executing: lalapps_PredictFstat --h0=9.999999999999999e-26 --cosi=1 --psi=0.39269908169872414 --Alpha=1.4596048908088417 --Delta=0.38422376285103965 --DataFiles='PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft;PyFstat_example_data/PyFstat_example_twoF_cumulative/L-843_L1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft' --minStartTime=1000000000 --maxStartTime=1000864000 --assumeSqrtSX=1e-23 --outputFstat=PyFstat_example_data/PyFstat_example_twoF_cumulative/PyFstat_example_twoF_cumulative.tmp --ephemEarth='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz' --ephemSun='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz'
11:05 ERROR   : Execution failed: Command 'lalapps_PredictFstat --h0=9.999999999999999e-26 --cosi=1 --psi=0.39269908169872414 --Alpha=1.4596048908088417 --Delta=0.38422376285103965 --DataFiles='PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft;PyFstat_example_data/PyFstat_example_twoF_cumulative/L-843_L1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft' --minStartTime=1000000000 --maxStartTime=1000864000 --assumeSqrtSX=1e-23 --outputFstat=PyFstat_example_data/PyFstat_example_twoF_cumulative/PyFstat_example_twoF_cumulative.tmp --ephemEarth='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz' --ephemSun='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz'' returned non-zero exit status 1.
11:05 ERROR   : ERROR: Failed to open matched file 'PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft'

XLAL Error - XLALSFTdataFind (SFTfileIO.c:277): I/O error
XLAL Error - InitPFS (PredictFstat.c:392): Check failed: (catalog = XLALSFTdataFind ( uvar->DataFiles, &constraints )) != ((void *)0)
XLAL Error - InitPFS (PredictFstat.c:392): Internal function call failed: I/O error
XLAL Error - main (PredictFstat.c:156): Check failed: InitPFS ( &GV, &uvar ) == XLAL_SUCCESS
XLAL Error - main (PredictFstat.c:156): Internal function call failed: I/O error

Traceback (most recent call last):
  File "PyFstat_example_twoF_cumulative.py", line 59, in <module>
    twoF = data.predict_fstat()
  File "/home/rodrigo/gravity/git/PyFstat/pyfstat/make_sfts.py", line 589, in predict_fstat
    twoF_expected, twoF_sigma = helper_functions.predict_fstat(
  File "/home/rodrigo/gravity/git/PyFstat/pyfstat/helper_functions.py", line 883, in predict_fstat
    run_commandline(cl_pfs)
  File "/home/rodrigo/gravity/git/PyFstat/pyfstat/helper_functions.py", line 303, in run_commandline
    out = subprocess.check_output(
  File "/usr/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'lalapps_PredictFstat --h0=9.999999999999999e-26 --cosi=1 --psi=0.39269908169872414 --Alpha=1.4596048908088417 --Delta=0.38422376285103965 --DataFiles='PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft;PyFstat_example_data/PyFstat_example_twoF_cumulative/L-843_L1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft' --minStartTime=1000000000 --maxStartTime=1000864000 --assumeSqrtSX=1e-23 --outputFstat=PyFstat_example_data/PyFstat_example_twoF_cumulative/PyFstat_example_twoF_cumulative.tmp --ephemEarth='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz' --ephemSun='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz'' returned non-zero exit status 1.
```

**Comment by Rodrigo-Tenorio** [comment]
This happens as well in other examples such as https://github.com/PyFstat/PyFstat/blob/893a575b48fa15e116d8d3ae1bf6b3bc7527fedc/examples/mcmc_examples/PyFstat_example_fully_coherent_MCMC_search.py when `Tsft = 1024`, but it works with `Tsft=400, 450, 900, 1800`.

**Comment by Rodrigo-Tenorio** [comment]
However, if duration is a multiple of `1024` there's no error involved, so this may be related to the "ending time" of the SFTs.

**Comment by Rodrigo-Tenorio** [comment]
Bingo:

Here is what the predict Fstat is looking for:
```
subprocess.CalledProcessError: Command 'lalapps_PredictFstat --h0=1e-24 --cosi=1.0 --psi=0.0 --Alpha=1.4596048908088417 --Delta=0.38422376285103965 --DataFiles='PyFstat_example_data/PyFstat_example_fully_coherent_MCMC_search/H-8437_H1_1024SFT_PyFstat_example_fully_coherent_MCMC_search-1000000000-8640000.sft' --minStartTime=1000000000 --maxStartTime=1008640000 --assumeSqrtSX=1e-23 --outputFstat=PyFstat_example_data/PyFstat_example_fully_coherent_MCMC_search/PyFstat_example_fully_coherent_MCMC_search.tmp --ephemEarth='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/earth00-40-DE405.dat.gz' --ephemSun='/home/rodrigo/gravity/venv/pyfstat/share/lalpulsar/sun00-40-DE405.dat.gz'' returned non-zero exit status 1.
```
But here is the actual produced file:
```
(pyfstat) [rodrigo@spinning-quadrupole mcmc_examples]$ ls PyFstat_example_data/PyFstat_example_fully_coherent_MCMC_search/
H-8438_H1_1024SFT_PyFstat_example_fully_coherent_MCMC_search-1000000000-8640512.sft  PyFstat_example_fully_coherent_MCMC_search.cff
```

**Comment by Rodrigo-Tenorio** [comment]
I'll leave this out of focus since it's probably just related to artificially produced data (since real data would always be an integer multiple of `Tsft`, as the number of SFTs is itself a whole number). Probably it's worth including a more explicit error message.

**Comment by dbkeitel** [comment]
Note how in the example output given above, the output
```
Successfully wrote SFTs to: PyFstat_example_data/PyFstat_example_twoF_cumulative/H-843_H1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft;PyFstat_example_data/PyFstat_example_twoF_cumulative/L-843_L1_1024SFT_PyFstat_example_twoF_cumulative-1000000000-864000.sft
```
from `Writer.run_makefakedata()` is already a lie.

So it seems that when we get the expected SFT names from `lalpulsar.OfficialSFTFilename`, we're rounding differently than MFD actually does it.

**Comment by dbkeitel** [comment]
Looks like I put a "close" hint into the wrong PR. Still being addressed in #205.
